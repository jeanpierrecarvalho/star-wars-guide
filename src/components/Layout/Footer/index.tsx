import Link from "next/link";

const Footer: React.FC = () => {
  const currentYear = new Date().getFullYear();

  return (
    <footer className="border-yellow-500 mt-12 pt-8 border-t-4 text-center">
      <p>
        Star Wars and all associated names and/or images are copyright{" "}
        <Link
          href="https://www.lucasfilm.com/"
          target="_blank"
          rel="noopener noreferrer"
          className="font-bold"
        >
          Lucasfilm Ltd.
        </Link>
      </p>
      <p className="mt-2 text-sm">
        Â© {currentYear}{" "}
        <Link
          href="https://jeanpierrecarvalho.pt/"
          target="_blank"
          rel="noopener noreferrer"
          className="font-bold"
        >
          Jean-Pierre Carvalho
        </Link>
        . All Rights Reserved.
      </p>
    </footer>
  );
};

export default Footer;
