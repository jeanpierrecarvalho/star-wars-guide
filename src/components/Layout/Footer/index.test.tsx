import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import Footer from '@/components/Layout/Footer'

describe('Footer Component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })
})
