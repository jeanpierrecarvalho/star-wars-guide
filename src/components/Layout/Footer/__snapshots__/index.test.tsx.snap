// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > matches the snapshot 1`] = `
<div>
  <footer
    class="border-yellow-500 mt-12 pt-8 border-t-4 text-center"
  >
    <p>
      Star Wars and all associated names and/or images are copyright
       
      <a
        class="font-bold"
        href="https://www.lucasfilm.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Lucasfilm Ltd.
      </a>
    </p>
    <p
      class="mt-2 text-sm"
    >
      Â© 
      2024
       
      <a
        class="font-bold"
        href="https://jeanpierrecarvalho.pt/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Jean-Pierre Carvalho
      </a>
      . All Rights Reserved.
    </p>
  </footer>
</div>
`;
